contracts/facets/ThirdPartyFacet.sol

```diff
@@ -18,61 +21,112 @@ contract ThirdPartyFacet {
+            revert LibErrors.SarcophagusCompromised(sarcoId);
+        }
+
+        // Confirm the sarcophagus is not buried
+        if (sarcophagus.resurrectionTime == 2 ** 256 - 1) {
+            revert LibErrors.SarcophagusInactive(sarcoId);
+        }
+
+        // Confirm the sarcophagus has not already been cleaned
+        if (sarcophagus.isCleaned) {
+            revert SarcophagusAlreadyCleaned(sarcoId);
+        }
+
+        // Confirm that the resurrectionTime + gracePeriod have passed
+        if (block.timestamp <= sarcophagus.resurrectionTime + s.gracePeriod) {
+            revert TooEarlyForClean(block.timestamp, sarcophagus.resurrectionTime + s.gracePeriod);
+        }
+
+        // Confirm tx sender is embalmer or admin
+        if (msg.sender != sarcophagus.embalmerAddress &&
+            msg.sender != LibDiamond.contractOwner()) {
+            revert SenderNotEmbalmerOrAdmin(msg.sender);
+        }
+
+        // if sender is embalmer, confirm current time is within embalmerClaimWindow
+        if (msg.sender == sarcophagus.embalmerAddress &&
+            block.timestamp > sarcophagus.resurrectionTime + s.gracePeriod + s.embalmerClaimWindow) {
+            revert EmbalmerClaimWindowPassed(block.timestamp, sarcophagus.resurrectionTime + s.gracePeriod + s.embalmerClaimWindow);
+        }
+
+        // if sender is admin, confirm embalmerClaimWindow has passed
+        if (msg.sender == LibDiamond.contractOwner() &&
+            block.timestamp <= sarcophagus.resurrectionTime + s.gracePeriod + s.embalmerClaimWindow) {
+            revert TooEarlyForAdminClean(block.timestamp, sarcophagus.resurrectionTime + s.gracePeriod + s.embalmerClaimWindow);
+        }
+
+        // sum of locked bonds and digging fees for all archaeologists that have failed to publish key shares before publish deadline and have not been accused
+        uint256 totalDiggingFeesAndLockedBonds = 0;
+        for (uint256 i = 0; i < sarcophagus.cursedArchaeologistAddresses.length; i++) {
+            LibTypes.CursedArchaeologist storage cursedArchaeologist = sarcophagus.cursedArchaeologists[sarcophagus.cursedArchaeologistAddresses[i]];
+            if (!cursedArchaeologist.isAccused && cursedArchaeologist.rawKeyShare.length == 0) {
+                totalDiggingFeesAndLockedBonds += cursedArchaeologist.diggingFee; // add digging fee for cursedArchaeologist
+                totalDiggingFeesAndLockedBonds += cursedArchaeologist.diggingFee; // add locked bond for cursedArchaeologist
+
+                // slash the archaeologist's locked bond for the sarcophagus
+                LibBonds.decreaseCursedBond(sarcophagus.cursedArchaeologistAddresses[i], cursedArchaeologist.diggingFee);
+            }
+        }
+
+        // if caller is embalmer, transfer them the total locked bonds and digging fees
+        if (msg.sender == sarcophagus.embalmerAddress) {
+            s.sarcoToken.transfer(sarcophagus.embalmerAddress, totalDiggingFeesAndLockedBonds);
```

Comment:
@katien -- do we need to handle the case where `totalDiggingFeesAndLockedBonds` is 0 here? Either all the archaeologists could have done their job, or there could be some combo of accused + successful unwraps which results in this being 0.

This could be calculated by the caller before calling this function (so could be gated that way). Another alternative is to have the `publishPrivateKey` calculate if there is no locked bond left (and set `isCleaned` to `true`), but I don't like the idea of making that function more expensive.

Reply:
I expect that to be gated client side because we're returning `hasLockedBond` from `getSarcophagus` so it's easy for clients to know whether or not they should be calling clean.
I don't think `publishPrivateKey` should set it `isCleaned` to true because cleaning is its own action with its own metrics. This comment did make me realize I'd forgotten to update `archaeologistCleanups` though so I will add that.

Maybe we shouldn't emit `Clean` and set `isCleaned` to true if `totalDiggingFeesAndLockedBonds` is 0. None of the metrics are getting updated if no archaeologist failed to do their duty and the embalmer and admin have no reason to call that function without a reward so not sure.

Reply:
yeah we could just return if no locked bond is available. in this scenario, a sarcophagus could be fully cleaned but not have `isCleaned` ever set to true, but that's fine since it's clean state can be derived in an alternate fashion.